ibis/omniscidb/operations.py

```diff
@@ -313,6 +313,123 @@ def extract_field_formatter(translator, expr, sql_attr=sql_attr):
     return extract_field_formatter
 
 
+def _n_weeks_year(year: ir.NumericValue):
+    """
+    Return the total number of weeks for the given year.
+
+    According to ISO 8601, a year can have 52 or 53 weeks.
+
+    The long years, with 53 weeks in them, occurs when:
+
+    * the year is a common year, starting on Thursday
+    * the year is a leap year, starting on Wednesday
+    * the year ends on Thursday
+    * the year is a leap year, ending on Friday
+    * years in which 1 January or 31 December are Thursdays
+
+    Otherwise, the year has 52 weeks.
+
+    Note: common year has 365 days and leap year has 366 days.
+
+    Parameters
+    ----------
+    year : ir.NumericValue
+
+    Returns
+    -------
+    ir.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+    """
+    # adjustment factor for week year extraction
+    def year_group(year: ir.NumericValue):
+        """
+        Return the year group cycle.
+
+        This group helps to check if it is 52 or 53 weeks.
+        For example, the year has 53 weeks if the year group is 4 or if
+        the group of the previous year is 3 (when the given year is leap).
+
```

Comment:
Can you add something like `This implementation is based on the formula in https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year`.

I find it a bit confusing, that in the above docstring you explain all the criteria for the number of weeks (year ends in Thursday...), and here you implement it in a total different way, without providing much context why.